crates/typst-library/src/text/quotes.rs

```diff
@@ -180,7 +212,24 @@ impl<'s> Quotes<'s> {
             "no" | "nb" | "nn" if alternative => low_high,
             "ru" | "no" | "nb" | "nn" | "ua" => ("’", "’", "«", "»"),
             _ if lang.dir() == Dir::RTL => ("’", "‘", "”", "“"),
-            _ => return Self::default(),
+            _ => default,
+        };
+
+        let quotes = quotes.as_ref();
+        let [single_open, single_close] = match quotes
+            .map(|q| q.single.as_ref())
+            .flatten()
+        {
+            Smart::Auto => [single_open, single_close],
+            Smart::Custom(QuoteSet { open, close }) => [open, close].map(|s| s.as_str()),
+        };
+
+        let [double_open, double_close] = match quotes
```

Comment:
this looks quite duplicate. could this be a function / method / something?

Reply:
I think for just one duplication it's fine, but I'll make it a closure or something.